hudi-common/src/main/java/org/apache/hudi/keygen/constant/KeyGeneratorType.java

```diff
@@ -33,31 +37,92 @@
+      + " a combination of fields. Similarly partition path can be configured to have multiple fields or only one field. "
+      + " This KeyGenerator expects value for prop \"hoodie.datasource.write.partitionpath.field\" in a specific format. "
+      + " For example: "
+      + " properties.put(\"hoodie.datasource.write.partitionpath.field\", \"field1:PartitionKeyType1,field2:PartitionKeyType2\").")
+  CUSTOM_AVRO_KEYGEN("org.apache.hudi.keygen.CustomAvroKeyGenerator"),
 
   @EnumFieldDescription("Simple Key generator for non-partitioned tables.")
-  NON_PARTITION,
+  NON_PARTITION_KEYGEN("org.apache.hudi.keygen.NonpartitionedKeyGenerator"),
+  @EnumFieldDescription("Simple Key generator for non-partitioned tables.")
+  NON_PARTITION_AVRO_KEYGEN("org.apache.hudi.keygen.NonpartitionedAvroKeyGenerator"),
 
   @EnumFieldDescription("Key generator for deletes using global indices.")
-  GLOBAL_DELETE;
+  GLOBAL_DELETE_KEYGEN("org.apache.hudi.keygen.GlobalDeleteKeyGenerator"),
+  @EnumFieldDescription("Key generator for deletes using global indices.")
+  GLOBAL_DELETE_AVRO_KEYGEN("org.apache.hudi.keygen.GlobalAvroDeleteKeyGenerator"),
+
+  @EnumFieldDescription("Automatic record key generation.")
+  AUTO_RECORD_KEYGEN("org.apache.hudi.keygen.AutoRecordGenWrapperKeyGenerator"),
+  @EnumFieldDescription("Automatic record key generation.")
+  AUTO_RECORD_AVRO_KEYGEN("org.apache.hudi.keygen.AutoRecordGenWrapperAvroKeyGenerator"),
+
+  @EnumFieldDescription("A KeyGenerator which use the uuid as the record key.")
+  UUID_KEYGEN("org.apache.spark.sql.hudi.command.UuidKeyGenerator"),
+
+  @EnumFieldDescription("Custom key generator for the Hudi table metadata.")
+  HOODIE_METADATA_KEYGEN("org.apache.hudi.metadata.HoodieTableMetadataKeyGenerator"),
+
+  @EnumFieldDescription("Meant to be used internally for the spark sql MERGE INTO command.")
+  MERGE_INTO_KEYGEN("org.apache.spark.sql.hudi.command.MergeIntoKeyGenerator"),
+
+  @EnumFieldDescription("Custom spark-sql specific KeyGenerator overriding behavior handling TimestampType partition values.")
+  SQL_KEYGEN("org.apache.spark.sql.hudi.command.SqlKeyGenerator");
+
+  private final String className;
+
+  KeyGeneratorType(String className) {
+    this.className = className;
+  }
+
+  public String getClassName() {
+    return className;
+  }
+
+  public static KeyGeneratorType fromClassName(String className) {
+    for (KeyGeneratorType type : KeyGeneratorType.values()) {
+      if (type.getClassName().equals(className)) {
+        return type;
+      }
+    }
+    throw new IllegalArgumentException("No KeyGeneratorType found for class name: " + className);
+  }
 
   public static List<String> getNames() {
     List<String> names = new ArrayList<>(KeyGeneratorType.values().length);
     Arrays.stream(KeyGeneratorType.values())
         .forEach(x -> names.add(x.name()));
     return names;
   }
+
+  public static String getKeyGeneratorClassName(HoodieConfig config) {
```

Comment:
Add annotation `@Nullable` to the method if there is any chance it could return null value, or return `Option` instead.